common/src/test/java/com/revenuecat/purchases/common/offlineentitlements/OfflineCustomerInfoCalculatorTest.kt

```diff
@@ -0,0 +1,210 @@
+        assertThat(receivedCustomerInfo?.entitlements?.all?.size).isEqualTo(2)
+
+        verifyEntitlement(receivedCustomerInfo, entitlementID, purchasedProduct)
+        verifyEntitlement(receivedCustomerInfo, secondEntitlementID, purchasedProduct)
+    }
+
+    @Test
+    fun `multiple products`() {
+        val entitlementID = "pro_1"
+        val secondEntitlementID = "pro_2"
+        val thirdEntitlementID = "pro_3"
+
+        val purchasedProducts = mockPurchasedProducts(
+            entitlementMap = mapOf(
+                "prod_1" to listOf(entitlementID),
+                "prod_2" to listOf(secondEntitlementID, thirdEntitlementID)
+            ),
+            expirationDate = testDatePlusOneDay
+        )
+
+        var receivedCustomerInfo: CustomerInfo? = null
+        offlineCustomerInfoCalculator.computeOfflineCustomerInfo(
+            appUserID,
+            { receivedCustomerInfo = it },
+            { fail("Should've succeeded") }
+        )
+        assertThat(receivedCustomerInfo).isNotNull
+
+        assertThat(receivedCustomerInfo?.activeSubscriptions).isEqualTo(listOf("prod_1", "prod_2").toSet())
+
+        assertThat(receivedCustomerInfo?.entitlements?.all?.size).isEqualTo(3)
+
+        val purchasedProduct = purchasedProducts.first { it.productIdentifier == "prod_1" }
+        verifyEntitlement(receivedCustomerInfo, entitlementID, purchasedProduct)
+        val secondPurchasedProduct = purchasedProducts.first { it.productIdentifier == "prod_2" }
+        verifyEntitlement(receivedCustomerInfo, secondEntitlementID, secondPurchasedProduct)
+        verifyEntitlement(receivedCustomerInfo, thirdEntitlementID, secondPurchasedProduct)
+    }
+
+    private fun verifyEntitlement(
+        receivedCustomerInfo: CustomerInfo?,
+        entitlementID: String,
+        purchasedProduct: PurchasedProduct
+    ) {
+        val receivedEntitlement = receivedCustomerInfo?.entitlements?.get(entitlementID)
+        assertThat(receivedEntitlement?.isActive).isTrue
+        assertThat(receivedEntitlement?.identifier).isEqualTo(entitlementID)
+        assertThat(receivedEntitlement?.productIdentifier).isEqualTo(purchasedProduct.productIdentifier)
+        assertThat(receivedEntitlement?.billingIssueDetectedAt).isNull()
+        assertThat(receivedEntitlement?.expirationDate).isEqualTo(testDatePlusOneDay)
+        assertThat(receivedEntitlement?.isSandbox).isFalse
+        assertThat(receivedEntitlement?.originalPurchaseDate).isEqualTo(testDate)
+        assertThat(receivedEntitlement?.latestPurchaseDate).isEqualTo(testDate)
+        assertThat(receivedEntitlement?.ownershipType).isEqualTo(OwnershipType.UNKNOWN)
+        assertThat(receivedEntitlement?.periodType).isEqualTo(PeriodType.NORMAL)
+        assertThat(receivedEntitlement?.store).isEqualTo(Store.PLAY_STORE)
+        assertThat(receivedEntitlement?.unsubscribeDetectedAt).isNull()
+    }
+
+    private fun mockPurchasedProducts(
```

Comment:
Maybe we can add a `// region helpers` or something like that for these 2 helper functions?